### PR TITLE
Part 7

### PR DESCRIPTION
**Pull request by vadimaganim**

### CODE REVIEW

File: ruby_oop_lessons/cargo_car.rb [hunk 1]
```diff
@@ -1,6 +1,19 @@
...
+    @capacity = capacity
+    @occupied_capacity = 0
   end
+
+  def took_capacity(desired_capacity)
+    if @capacity - desired_capacity >= 0
+      @capacity -= desired_capacity
+      @occupied_capacity = desired_capacity
+      "You occupied #{desired_capacity}"
+    else
```
Comment (IvanNosov) [inline ruby_oop_lessons/cargo_car.rb]: elsif?

File: ruby_oop_lessons/cargo_car.rb [hunk 2]
```diff
@@ -1,6 +1,19 @@
...
+      raise "Sorry, but you want to occupied more than we have: #{@capacity} Try again!" if @capacity - desired_capacity < 0
```
Comment (IvanNosov) [inline ruby_oop_lessons/cargo_car.rb]: line is to long, please use rubocop

Comment (IvanNosov) [reply ruby_oop_lessons/cargo_car.rb]: also `@capacity - desired_capacity < 0` won't show you a proper capacity

File: ruby_oop_lessons/main.rb [hunk 3]
```diff
@@ -120,6 +124,51 @@ def self.play_again?
...
+      occupied_capacity = gets.chomp.to_i
+      p @train[t_index].cars[car_index].took_capacity(occupied_capacity)
+    rescue RuntimeError => e
+      p e.message
+      retry
+    end
+    p "Capacity was occupied! left: #{@train[t_index].cars[car_index].capacity}"
+  end
+
+  def self.tooking_seat
```
Comment (IvanNosov) [inline ruby_oop_lessons/main.rb]: take

File: ruby_oop_lessons/passenger_car.rb [hunk 4]
```diff
@@ -1,6 +1,18 @@
...
-  def initialize
+  attr_reader :car_type, :seats, :occupied_seats
+  def initialize(seats)
     @car_type = :passenger
+    @seats = seats
+    @occupied_seats = 0
+  end
+
+  def took_seat
+    if @seats != 0
```
Comment (IvanNosov) [inline ruby_oop_lessons/passenger_car.rb]: if seats will be -50 it will return true

File: ruby_oop_lessons/main.rb [hunk 5]
```diff
@@ -120,6 +124,51 @@ def self.play_again?
...
+      p @train[t_index].cars[car_index].took_seat
+    rescue RuntimeError => e
+      p e.message
+      retry
+    end
+    p "Seat was occupied! left: #{@train[t_index].cars[car_index].seats}"
+  end
+
+  def self.station_block
+    Station.checker(@stations) { |element| puts element.trains }
```
Comment (IvanNosov) [inline ruby_oop_lessons/main.rb]: |element| is not informative

File: ruby_oop_lessons/main.rb [hunk 6]
```diff
@@ -120,6 +124,51 @@ def self.play_again?
...
+  end
+
+  def self.train_block
+    train_index
+    p 'Enter train index which you choose: '
+    t_index = gets.chomp.to_i
+    Train.checker(@train[t_index]) { |element| puts element.cars }
```
Comment (IvanNosov) [inline ruby_oop_lessons/main.rb]: |element| is not informative

File: ruby_oop_lessons/station.rb [hunk 7]
```diff
@@ -12,8 +12,13 @@ def initialize(name)
     @@instances << self
   end
 
+  def self.checker(stations, &block)
+    stations.each { |station| block.call(station) }
+  end
+
   def validate!
     raise 'Name can\'t be nil' if name.empty?
+
```
Comment (IvanNosov) [inline ruby_oop_lessons/station.rb]: why did you add this line?

File: ruby_oop_lessons/train.rb [hunk 8]
```diff
@@ -16,10 +16,15 @@ def initialize(number)
...
 
+  def self.checker(train, &block)
+    block.call(train)
+  end
+
   def validate!
     raise 'Number can\'t be nil' if number.nil?
     raise 'Number shoud be at least 6 symbols' if number.length < 5
     raise 'Number has invalid format' if number !~ TRAIN_NUMBER
+
```
Comment (IvanNosov) [inline ruby_oop_lessons/train.rb]: why did you add this line?

File: ruby_oop_lessons/cargo_car.rb [hunk 9]
```diff
@@ -1,6 +1,20 @@
...
+    @capacity = capacity
+    @occupied_capacity = 0
+  end
+
+  def take_capacity(desired_capacity)
+    if @capacity - desired_capacity >= 0
+      @capacity -= desired_capacity
+      @occupied_capacity += desired_capacity
+      "You occupied #{desired_capacity}"
+    elsif (@capacity - desired_capacity).negative?
```
Comment (IvanNosov) [inline ruby_oop_lessons/cargo_car.rb]: leave just `else`

Comment (vadimaganim) [reply ruby_oop_lessons/cargo_car.rb]: @IvanNosov Fixed:)
